src/types.ts

```diff
@@ -0,0 +1,182 @@
+import { Cookie, Headers, HttpMethod, Page, ResourceType, Viewport } from 'puppeteer'
+// import { Viewport } from 'puppeteer/DeviceDescriptors'
+
+interface IAuthlessCore {
+  name: string
+  domainPathRouter: IDomainPathRouter
+}
+
+interface IBotRouter {
+  botMap: {[url: string ]: IBot[]}
+  botIndices: {[url: string ]: number}
+  getBotForUrl(url: string)
+  // getBot(): IBot | undefined
+  getBotByUsername(username: string): IBot | undefined
+}
+
+interface IBot {
+  username: string
+  password: string
+  hitCount: number
+  loginCount: number
+  captchaCount: number
+  foundLogin: (found: Boolean) => void
+  foundCaptcha: (found: Boolean) => void
+  getHitCount: () => number
+  getLoginHitCount: () => number
+  getCaptchaHitCount: () => number
+}
+
+interface IResponseMeta {
+  url: string
+  viewport: Viewport
+  title: string
+}
+
+interface IResponse {
+  meta: IResponseMeta
+  page: any
+  content: any
+  cookies: Cookie[]
+  xhrs: Xhr[]
+}
+
+interface IDomainPathRouter {
+  domainMap: {[url: string ]: IDomainPath}
+  getDomainPathForUrl: (url: string) => IDomainPath | undefined
+}
+
+interface IDomainPath {
+  domain: string
```

Comment:
Is there a reason why this is called `domain`? Is this the `name`? I'm not quite sure of what `domain` means here.

Reply:
Something to show when logging so what we know which domainPath is doing the logging